Fix 226

anthonydahanne opened the pull request:
## Summary
Support new env. variable BP_MAVEN_ACTIVE_PROFILES
* append its profiles last


Support new env. variable BP_MAVEN_ADDITIONAL_BUILD_ARGUMENTS
* append its arguments after the BP_MAVEN_BUILD_ARGUMENTS

## Use Cases
```
pack build -e BP_MAVEN_ACTIVE_PROFILES=native,-dev -e BP_MAVEN_ADDITIONAL_BUILD_ARGUMENTS=-DskipJavadoc -p ~/workspaces/paketo-buildpacks/samples/java/maven/ plop -b paketo-buildpacks/syft \
-b paketo-buildpacks/bellsoft-liberica \
-b . \
-b paketo-buildpacks/executable-jar \
 --builder paketobuildpacks/builder:base --verbose --clear-cache

[...]

Paketo Buildpack for Maven {{.version}}
  https://github.com/paketo-buildpacks/maven
  Build Configuration:
    $BP_EXCLUDE_FILES                                                                            colon separated list of glob patterns, matched source files are removed
    $BP_INCLUDE_FILES                                                                            colon separated list of glob patterns, matched source files are included
    $BP_JAVA_INSTALL_NODE                 false                                                  whether to install Yarn/Node binaries based on the presence of a package.json or yarn.lock file
    $BP_MAVEN_ACTIVE_PROFILES             native,-dev                                            the active profiles (comma separated: such as: p1,!p2,?p3) to pass to Maven
    $BP_MAVEN_ADDITIONAL_BUILD_ARGUMENTS  -DskipJavadoc                                          the additionnal arguments (appended to BP_MAVEN_BUILD_ARGUMENTS) to pass to Maven
    $BP_MAVEN_BUILD_ARGUMENTS             -Dmaven.test.skip=true --no-transfer-progress package  the arguments to pass to Maven
    $BP_MAVEN_BUILT_ARTIFACT              target/*.[ejw]ar                                       the built application artifact explicitly.  Supersedes $BP_MAVEN_BUILT_MODULE
    $BP_MAVEN_BUILT_MODULE                                                                       the module to find application artifact in
    $BP_MAVEN_DAEMON_ENABLED              false                                                  use maven daemon
    $BP_MAVEN_POM_FILE                    pom.xml                                                the location of the main pom.xml file, relative to the application root
    $BP_MAVEN_SETTINGS_PATH                                                                      the path to a Maven settings file
    $BP_MAVEN_VERSION                     3                                                      the Maven version
    $BP_NODE_PROJECT_PATH                                                                        configure a project subdirectory to look for `package.json` and `yarn.lock` files
    Creating cache directory /home/cnb/.m2
  Compiled Application: Contributing to layer
    Executing mvnw --batch-mode -Dmaven.test.skip=true --no-transfer-progress package -DskipJavadoc -P native,-dev
      [INFO] Scanning for projects...
[...]

```

## Checklist
<!-- Please confirm the following -->
* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

dmikusa commented:
This looks good. The README.md should have entries for the new Config settings.

A reminder and it needs a separate PR, we'll want to add these new settings to the docs as well.

anthonydahanne commented:
> This looks good. 

Thanks!

> The README.md should have entries for the new Config settings.

Updated and added in this PR: https://github.com/paketo-buildpacks/paketo-website/pull/671

dmikusa commented on README.md:
```diff
@@ -34,19 +34,22 @@ The buildpack will do the following:
+|----------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| `$BP_MAVEN_VERSION`                    | Configure the major Maven version (e.g. `3`, `4`).  Since the buildpack only ships a single version of each supported line, updates to the buildpack can change the exact version of Maven installed. If you require a specific minor/patch version of Maven, use the Maven wrapper instead.                                                                         ...
+| `$BP_MAVEN_BUILD_ARGUMENTS`            | Configure the arguments to pass to Maven.  Defaults to `-Dmaven.test.skip=true --no-transfer-progress package`. `--batch-mode` will be prepended to the argument list in environments without a TTY.                                                                                                                                                                 ...
+| `$BP_MAVEN_BUILD_ARGUMENTS`            | Configure the arguments to pass to Maven.  Defaults to `-Dmaven.test.skip=true --no-transfer-progress package`. `--batch-mode` will be prepended to the argument list in environments without a TTY.                                                                                                                                                                 ...
```
I think this line is a duplicate?

dmikusa commented:
Just curious, are you planning to submit a PR for gradle that adds `BP_GRADLE_BUILD_ARGUMENTS`? That would seem to be a reasonable addition. Not sure if the other env variable makes sense for gradle?
